lib/cache/index.ts

```diff
@@ -0,0 +1,67 @@
+import { CacheOptions, DEFAULT_SCOPE, DEFAULT_OPTIONS } from './CacheOptions';
+import { factoryCacheService as factoryCacheService } from './FactoryCacheService';
+import { Cache } from './Cache';
+import { StorageType } from './StorageType';
+import { ClassType } from '../interfaces/class';
+import { checkCacheOptions } from './checkCacheOptions';
+
+export { CacheOptions };
+
+/**
+ * Caches the result of a method.
+ * @param timeout cache timeout in milliseconds.
+ * @param options (optional) caching options.
+ */
+export function cache(timeout: number, options: CacheOptions = DEFAULT_OPTIONS) {
+  checkCacheOptions(options);
+  const scope = options ? options.scope : DEFAULT_SCOPE;
+
+  return function (target: any, propertyKey: any, descriptor: PropertyDescriptor) {
+    const method = descriptor.value;
+    const storage =
+      scope === 'class'
```

Comment:
Again, [SRP](https://github.com/labs42io/clean-code-typescript#single-responsibility-principle-srp) violated. The `scope` should be abstracted, see the implementation of `throttle` and the submitted pull request for `bulkhead` from @vyrtus15 .